docs/rfcs/0004-suggested-excluded-apps-tech-spec.md

```diff
@@ -0,0 +1,139 @@
+-   (1 point) Sort recommended section
+-   (1 point) Sort "others" section
+-   (1-2 points) Update sort when something is added or removed
+-   (1-2 points) Ensure search works, shows one section, is alphabetically sorted
+-   (2-5 points) Ensure accessibility is up to date
+-   (1 point) Add metrics for how many suggested (and non-suggested) apps are excluded\
+
+### Day 2
+
+These are important components that we are committed to delivering, but are outside the scope of this epic. They are items that will be done as soon as the epic is closed, but do not block the feature on shipping.
+
+-   Engineering metrics
+    -   Metrics will be created for a successful deep linking, as well as an erroneous deep linking (the app cannot figure out how to resolve the deep link).
+    -   Add performance metrics around load time of the Excluded Apps screen.
+    -   Create a metric for when the daily job was successful, and another one for when it fails.
+    -   A metric should be recorded when the notification is sent
+
+Other options considered
+------------------------
+
+Design and research went through several iterations, so the [designs](https://www.figma.com/file/UZYzma7hlcfE5ke3z8jGbN/App-exclusions-suggestions?type=design&node-id=196-6366&mode=design&t=RL1hdfBQLMS1rKVa-0) were set before this tech spec was written. (See [completed tickets in the epic](https://mozilla-hub.atlassian.net/browse/VPN-4412) for more background.)
+
+Other options considered included dynamically serving the suggested lists from addons or our repo or Guardian, each of which added complexity for hosting/validating. With the rarity of the list being updated (and lack of time-sensitivity when it does), this draft bundles the lists with the app. Hosting the list in Guardian was difficult (which is a private repo, so the list would need to be hosted elsewhere to allow public suggestions).
+
+### Other options considered for suggested app files
+
+The goals for the file format:
+-   Easy to read by humans
+-   Easy for humans to add to, no matter their level of technical proficiency
+-   Easy for us to parse in the app
+-   Allowing comments in the file. While we still could structure a text file to allow comments, this became less important when the plan became "one file per platform, one app per line".
+
+Initially, there was one additional goal (no longer consideration):
+-   To have one file for all platforms, so that we'd remember to add a specific app to all platforms at once. With a conversation around potentially wanting different suggested app categories on mobile, this seems unimportant.
+
+Other options considered included CSV, YAML, and JSON. While YAML could be useful (as it's easy for less-technical people to read), we don't have a good YAML parser in the app yet. We could add a library, of course. However, we'd need to write our own document format test, because the YAML file (or JSON or CSV) would need to be structured in a specific way.
```

Comment:
Qt provides a JSON parser out of the box. Using JSON would allow us to use JSON Schema to define the format of the file and would make it very easy to test. That would also not be anything new for the codebase, we currently do exactly that for addon manifests and the schema check is run on the linter CI job already. See https://github.com/mozilla-mobile/mozilla-vpn-client/blob/main/.github/workflows/linters.yaml#L43-L45.

Reply:
Thanks, I didn't know about that. I've updated this statement to be more accurate.